### ISSUE TITLE
Define new Zowe components dependency json file

### ISSUE THREAD

**Issue by jackjia-ibm** [zowe/zowe-install-packaging#498]
The new json file will be used to generate current `artifactory-download-spec.json.template`. After the new json file is in place, we can retire artifactory-download-spec.json.template.

The new json file will also be used to pick components tags. So we can use this file to generate source build, or license scanning, or potential compare changes between releases.

The new json file should be able to define:
- component/sub-component ID
- which repository (release or snapshot)
- exact version
- patch level versions following pattern defined in https://docs.npmjs.com/about-semantic-versioning#using-semantic-versioning-to-specify-update-types-your-package-can-accept
- artifact name pattern, since we may have several artifacts uploaded to the same version folder

This json may be merged into `manifest.json` as `dependencies` section.

**Comment by jackjia-ibm** [comment]
Dependent task: https://github.com/zowe/jenkins-library/issues/18

**Comment by jackjia-ibm** [comment]
Discussion record:

From Mark:

```
{
    "source_tags" : {
        "explorer-api-common": "0.3.1",
        "data-sets": "0.2.0",
        "jobs": "0.2.2",
        "imperative": "v2.4.1",
        "zowe-cli-cics-plugin": "v1.0.0"
    },
    "binary_artifacts" : {
        "apiml" : "//path/to/artifactory/zip"

    },
    "__meta__" : {
        "dependencyDecisionsYaml" : "https://raw.github.....",
        "zoweVersion"  : "1.4.0"
    },
   "componentMap" : {
    "Zowe API Mediation Layer": ["api-layer", "api-layer-api-catalog-ui-frontend"],
    "Zowe Explorer Data Sets REST Server": ["data-sets"],
    "Zowe Desktop Database Browser Plug-in": ["db-browser-db2-nodeServer", "db-browser-webClient"]
}
}
```

From Jack:
```
  "binaryDependencies": {
    "org.zowe.zlux.zlux-core": {
      // pick exact artifact with path
      "artifact": "libs-snapshot-local/org/zowe/zlux/zlux-core/1.3.0-RC/zlux-core-1.3.0-20190607.143930.pax"
    },
    "org.zowe.explorer-jes": {
      // still pick exact artifact by release version
      // match 0.0.21 in libs-release-local/org/zowe/explorer-jes/0.0.21/*
      // and it should only have one artifact in the folder
      "version": "0.0.21"
    },
    "org.zowe.explorer-mvs": {
      // pick a most recent build
      // match 0.0.* from libs-snapshot-local/org/zowe/explorer-mvs/0.0.*/*
      "version": "~0.0.21",
      // optional to specify which artifact to pick from, if more than one in the folder
      // with this option, will try to find the artifact with pattern libs-snapshot-local/org/zowe/explorer-mvs/0.0.*/explorer-mvs-special-name-*.pax
      "artifact": "explorer-mvs-special-name-*.pax"
    },
    "org.zowe.another": {
      // pick a most recent build
      // match 0.* from libs-snapshot-local/org/zowe/another/0.*/*
      "version": "^0.0.21",         // match 0.0.* in libs-snapshot-local,
    },
    "license": {
      "artifact": "libs-release-local/org/zowe/licenses/1.0.0/zowe_licenses_full.zip"
    },
    // another way to define with same effect
    "org.zowe.licenses": {
      "version": "1.0.0"
    }
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/zowe"
  },
  "sourceDependenciesClaim": "Source dependecies are only valid for formal release. For nightly builds or branch builds, these values may not reflect the real source be included into the build.",
  "sourceDependencies": {
    "explorer-api-common": {
      "tag": "0.3.1",                          // we can also use this to render changelist for coming release
      "componentGroup": "Zowe Explorer APIs"   // used in generating tpsr
    },
    "another-repo": {
      "commit": "0b463274fa3d3f1dbdfad2d9a7aa79f5894a984d",
      "componentGroup": "Zowe Explorer APIs",
      "admins": [{
         name: "Admin A",
         email: "a1@gmail.com"
      }]
    }
  }
```

**Comment by MarkAckert** [comment]
I created a branch `stagings/new-manifest` to work on this in coordination with the dependency scanning pipeline; I am following a format similar to your proposal and will iterate on that.

**Comment by MarkAckert** [comment]
Ready for merge, PR here: https://github.com/zowe/zowe-install-packaging/pull/543

**Comment by jackjia-ibm** [comment]
Pending question: if we can define an universal way to store scripts of how to initialize each sub-project, like some of them using npm install, some of them using gradle build, etc. We may need some assumptions on how to handle credentials.

**Comment by jackjia-ibm** [comment]
Since main features related to this are already implemented, will leave #555 open and close this.
